cmd/api.go

```diff
@@ -378,21 +386,91 @@ var get = &cobra.Command{
 }
 
 var getTX = &cobra.Command{
-	Use:              "tx <entryhash>",
+	Use:              "tx <txid>",
 	Short:            "Fetch the transaction by the given entryhash",
 	PersistentPreRun: always,
 	PreRun:           SoftReadConfig,
 	Args:             cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		ehash := factom.NewBytes32FromString(args[0])
-		if ehash.IsZero() {
-			cmd.PrintErrf("entryhash must be a 64 character hex string")
+		_, _, err := pegnet.SplitTxID(args[0])
+		if err != nil {
+			cmd.PrintErrf("txid is invalid: %s", err.Error())
+		}
+
+		cl := srv.NewClient()
+		cl.PegnetdServer = viper.GetString(config.Pegnetd)
+		var res srv.ResultGetTransactions
+		err = cl.Request("get-transaction", srv.ParamsGetPegnetTransaction{TxID: args[0]}, &res)
+		if err != nil {
+			fmt.Printf("Failed to make RPC request\nDetails:\n%v\n", err)
+			os.Exit(1)
+		}
+
+		data, err := json.Marshal(res)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Println(string(data))
+	},
+}
+
+var getTXs = &cobra.Command{
+	Use:   "txs <entryhash | FA address | height>",
+	Short: "Fetch all transactions for an entryhash, FA address, or height",
+	Long: "Fetch all transactions for an entryhash, FA address, or height. " +
+		"If a --burn, --cvt, --tran, or --coin is provided, then only the flags" +
+		" provided will be displayed.",
+	Example:          "pegnetd txs 07cebdd5d3f5216f36f792d71f030af07ddaa99147929d9af477833ee4c586a5",
+	PersistentPreRun: always,
+	PreRun:           SoftReadConfig,
+	Args:             cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var height int
+		// determine the params
+		var params srv.ParamsGetPegnetTransaction
+
+		// An entryhash?
+		bytes, err := hex.DecodeString(args[0])
+		if err == nil && len(bytes) == 32 {
+			params.Hash = args[0]
+			goto FoundParams
+		}
+
+		// A factoid address maybe?
+		_, err = factom.NewFAAddress(args[0])
+		if err == nil {
+			params.Address = args[0]
+			goto FoundParams
+		}
+
+		// Ok, maybe it's a height!
+		height, err = strconv.Atoi(args[0])
+		if err == nil {
+			params.Height = height
+			goto FoundParams
+		}
+
+		// I give up.
+		cmd.PrintErrf("param invalid. could not determine type")
+	FoundParams:
+
+		params.Conversion, _ = cmd.Flags().GetBool("cvt")
+		params.Burn, _ = cmd.Flags().GetBool("burn")
+		params.Transfer, _ = cmd.Flags().GetBool("tran")
+		params.Coinbase, _ = cmd.Flags().GetBool("coin")
+		oneSet := params.Conversion || params.Burn || params.Transfer || params.Coinbase
+		if !oneSet {
```

Comment:
setting all to true is the same thing as setting all to false on the api side, so this is unnecessary. the json format has `omitempty` tags so that you don't even have to include any of them

Reply:
Ah ok. I will change it to only include if the flags are set.